pkg/kubelet/kuberuntime/instrumented_services_test.go

```diff
@@ -41,7 +48,8 @@ func TestRecordOperation(t *testing.T) {
 	prometheusURL := "http://" + l.Addr().String() + "/metrics"
 	mux := http.NewServeMux()
 	//lint:ignore SA1019 ignore deprecated warning until we move off of global registries
-	mux.Handle("/metrics", legacyregistry.Handler())
+	handler := promhttp.InstrumentMetricHandler(prometheus.DefaultRegisterer, promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
```

Comment:
We don't really need an InstrumentedMetricHandler here since we aren't testing handler specific metrics coming from the Prometheus client. As such, having only the handler for the registry should be enough.

Reply:
Hi Damien, 
it failed with `make test KUBE_RACE=-race KUBE_TIMEOUT=--timeout=600s GOFLAGS=-count=10 WHAT=./pkg/kubelet/kuberuntime/` before.

Since using local registry, the handler become `promhttp.InstrumentMetricHandler(prometheus.DefaultRegisterer, promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))`.

If not using local registry, do you know other ways could fix `make test KUBE_RACE=-race KUBE_TIMEOUT=--timeout=600s GOFLAGS=-count=10 WHAT=./pkg/kubelet/kuberuntime/`? Thanks!

Reply:
I was not saying that we should move away from the local registry, this is definitely the way to fix the issue at hand, but I was saying that we shouldn't need an instrumented handler here. I am suggesting the following change:
```suggestion
	handler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
```

Reply:
Got it. Thanks Damien! 
I will try.